cl/corpus_importer/tasks.py

```diff
@@ -3174,3 +3177,105 @@ def classify_case_name_by_llm(self, cluster_pk: int, recap_document_id: int):
             "Case names successfully updated https://www.courtlistener.com/opinion/%s/decision/",
             cluster.id,
         )
+
+
+@app.task(
+    bind=True,
+    autoretry_for=(
+        ConnectionError,
+        Timeout,
+    ),
+    max_retries=5,
+    ignore_result=True,
+)
+def download_qp_scotus_pdf(self, docket_id: int) -> None:
+    """Download and store the SCOTUS 'Questions Presented' PDF for a docket.
+
+    This task looks up the Docket and its ScotusDocketMetadata, extracts the
+    questions_presented_url, validates that the response is a PDF, and stores
+    it in the questions_presented_file field.
+
+    :param: docket_id: The related Docket ID
+    :return: None
+    """
+    try:
+        docket = Docket.objects.get(pk=docket_id)
+    except Docket.DoesNotExist:
+        logger.warning(
+            "SCOTUS PDF download: Docket %s does not exist; skipping.",
+            docket_id,
+        )
+        return
+
+    scotus_meta = ScotusDocketMetadata.objects.get(docket=docket)
+
+    qp_url = scotus_meta.questions_presented_url
+    if not qp_url:
+        logger.info(
+            "SCOTUS PDF download: No questions_presented_url for docket %s.",
+            docket_id,
+        )
+        return
+
+    # Avoid re-downloading if we already have a file.
+    if scotus_meta.questions_presented_file:
+        logger.info(
+            "SCOTUS PDF download: questions_presented_file already present "
+            "for docket %s; skipping.",
+            docket_id,
+        )
+        return
+
+    logger.info(
+        "SCOTUS PDF download: Fetching Questions Presented PDF for docket %s "
+        "from %s",
+        docket_id,
+        qp_url,
+    )
+    try:
+        response = requests.get(
+            qp_url,
+            stream=True,
+            timeout=60,
+            headers={"User-Agent": "Free Law Project"},
+        )
+        response.raise_for_status()
+    except RequestException as exc:
+        logger.warning(
+            "SCOTUS PDF download: Unable to download %s for docket %s. "
+            "Exception was: %s",
+            qp_url,
+            docket_id,
+            exc,
+        )
+        return
+
+    if not is_pdf(response):
+        logger.warning(
+            "SCOTUS PDF download: Expected application/pdf for docket %s "
+            "from %s; aborting.",
+            docket_id,
+            qp_url,
+        )
+        return
+
+    with NamedTemporaryFile(prefix="scotus_qp_", suffix=".pdf") as tmp:
+        # Download the PDF into a tmp file to avoid using too much memory
+        for chunk in response.iter_content(chunk_size=8 * 1024):
+            if chunk:
+                tmp.write(chunk)
+
+        tmp.flush()
+        tmp.seek(0)
+
+        filename = f"{docket_id}-qp.pdf"
+        scotus_meta.questions_presented_file.save(
+            filename,
+            File(tmp),
+            save=True,
+        )
```

Comment:
Should we explicitly close the response connection after writing the file? 

I’ve usually seen `stream=True` used inside a context manager like `with requests.get(...) as response: ` so that the connection is properly closed once we're done reading from it. In this function, we call requests.get without a context manager

Reply:
You're right! Thanks for catching this. I've added a context manager to handle this request.